src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java

```
@@ -371,4 +369,22 @@ private static void delay(long delayInMs) {
         }
     }
 
+    private Decision waitPrecondition(long offset) {
+        Decision decision = Precondition.Decision.WAIT;
+        long endTime = System.currentTimeMillis() + PRECONDITION_TIMEOUT * 1000;
+        while (decision == Decision.WAIT && System.currentTimeMillis() < endTime) {
+            decision = precondition.canProcess(subAgentName, offset);
+            if (decision == Decision.WAIT) {
+                try {
+                    Thread.sleep(100);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    return Decision.SKIP;
```

We should throw here instead of returning an arbitrary decision.

---

I agree

---

I first fixed with returning WAIT but actually throwing PreConditionTimeoutException might make more sense. WDYT?